src/lib/esnext.array.d.ts

```diff
@@ -0,0 +1,17 @@
+interface ArrayConstructor {
+    /**
+     * Creates an array from an async iterator or iterable object.
+     * @param iterableOrArrayLike An async iterator or array-like object to convert to an array.
+     */
+    fromAsync<T>(iterableOrArrayLike: AsyncIterable<T> | Iterable<T | Promise<T>> | ArrayLike<T | Promise<T>>): Promise<T[]>;

```

Comment:
from https://github.com/microsoft/TypeScript/issues/50803#issuecomment-1249815267, I think the `Promise`-typed parameters should be `PromiseLike`s.

Reply:
Should it also return a `Promise`?

Reply:
I could be wrong, but I think the return type of built-in method is Promise even if it accepts PromiseLikes.